### ISSUE TITLE
FORCE MMQ & TENSOR CORES set to "maybe" when loading a model.

### ISSUE THREAD

**Issue by Nexesenex** [LostRuins/koboldcpp#556]
When I load a model (e.g Yi 34b llamafied) with cublas mmq with KCPP, I get this:

"ggml_init_cublas: GGML_CUDA_FORCE_MMQ: maybe" (!!! I guess yes)
"ggml_init_cublas: CUDA_USE_TENSOR_CORES: maybe" (!!! I don't know)

A simple question : why doesn't it say yes or no, instead of maybe?
And what is the actual answer in KCPP with untouched CMakeLists.txt and ggml-cuda.cu used to make the koboldcpp_cublas.dll?

**Comment by LostRuins** [comment]
So the reason why I set it to "maybe" is because the utilization of MMQ in KCPP follows a bit of a different logic from upstream since 1.48

```
'Tensor Core' Changes: KoboldCpp now handles MMQ/Tensor Cores differently from upstream. Here's a breakdown:
old approach (everybody): if mmq is enabled, just use mmq. If cublas is enabled, just use cublas. MMQ dimensions set to "FAVOR BIG"
new approach (upstream llama.cpp): you cannot toggle mmq anymore. It is always enabled. MMQ dimensions set to "FAVOR SMALL". CuBLAS always kicks in if batch > 32.
new approach (koboldcpp): you CAN toggle MMQ. It is always enabled, until batch > 32, then CuBLAS only kicks in if MMQ flag is false, otherwise it still uses MMQ for all batches. MMQ dimensions set to "FAVOR BIG".
```

I realize that such debug output is kind of useless, so I will be removing the "maybe" statements.

CMakeLists is intended for building for CuBLAS, but does not support the other make targets. I'm not sure what you mean by "untouched"

So in summary If you're wondering, MMQ batch processing cannot be fully disabled, but can be toggled on or off for large batch sizes to use cublas instead. And yes, the tick mark does have an effect, you will be able to see it based on the number of layers offloadedable.

The next version does include the per-layer KV stuff to complicate matters, but that's a different issue.

**Comment by Nexesenex** [comment]
By untouched I mean "not modified by the user who use it to compile" lol. Thanks for the clarification.

I'm still using your latest experimental before the KV per layer merge for full offloads except Mixtral, because I noticed that this novelty eats more VRAM without a bonus on the performances for a full offload (and both the full and the partial offload might not be optimal on multi-GPU solution because the KV cache is stored in GPU0 VRAM even if some layers are on the other GPUs, thus while retaining the hassle of the tensor split proper configuration, especially on heterogenous bi-GPU configurations).

I really liked Johannes Gaessler work over KVQ8_0, I merged it in late september in a KCPP experimental build and it was working somehow, albeit imperfectly (No dual-GPU I could make work, no CodeLlama 34b, crash over regeneration of answer).
